### ISSUE TITLE
TypeScript: `moduleResolution: "node16"` leads to broken types

### ISSUE THREAD

**Issue by zipme** [amannn/next-intl#347]
### Description

When importing functions from `next-intl/server` package in a server component (e.g. app/[locale]/page.tsx) it shows incorrect typing. It only happens if tsconfig.json's `moduleResolution` is set to `"node16"` or `nodenext`

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://codesandbox.io/p/sandbox/next-intl-bug-template-app-forked-wgnkxs?file=%2Fsrc%2Fapp%2F%5Blocale%5D%2Fpage.tsx%3A4%2C1-5%2C1

### Reproduction description

// app/[locale]/page.tsx
```tsx
import { getTranslations } from "next-intl/server";

export default async function Home() {
  const t = await getTranslations("index"); // <- This expression is not callable. Type 'void' has no call signatures.ts(2349)

  return (
    <>
      <div>Test locale {t("Hello world")}</div>
    </>
  );
}

```

### Expected behaviour

`getTranslations` should give correct typing

**Comment by amannn** [comment]
Hmm, interesting. Can I ask what your use case is that made you change `moduleResolution` from `node` to `node16`?

**Comment by amannn** [comment]
I've added a note about this to https://github.com/amannn/next-intl/issues/342, I hope that can help to resolve this issue. Thanks for reporting!

**Comment by amannn** [comment]
I probably have a fix for this in https://github.com/amannn/next-intl/pull/371 that can be released independently of #342!
